typecats/tc.py

```diff
@@ -227,15 +234,22 @@ def try_struc_cat(d: ty.Optional[StrucInput]) -> ty.Optional[C]:
 
 
 def set_unstruc_converter(
-    cls: ty.Type[C],
-    converter: cattr.Converter = _TYPECATS_DEFAULT_CONVERTER,
-    strip_defaults_converter: cattr.Converter = get_stripping_converter(),
+    cls: ty.Type[C], converter: cattr.Converter = _TYPECATS_DEFAULT_CONVERTER
 ):
     """If you want to change your mind about the built-in Converter that
     is meant to run when you call the object method YourCatObj.unstruc(), you
     can reset it here. By default, it is defined by the converter
     keyword argument on the Cat decorator.
 
     """
-    _unstruc = make_unstruc(converter, strip_defaults_converter)
-    setattr(cls, UNSTRUCTURE_NAME, _unstruc)
+    setattr(cls, UNSTRUCTURE_NAME, make_unstruc(converter))
+
+
+def unstruc_strip_defaults(obj: ty.Any) -> ty.Any:
+    """A functional-ish interface for stripping defaults.
+
+    Note that if you need to use a specific converter,
+    you'll want to dig in and set this context directly.
+    """
+    with stack_context(ShouldStripDefaults, True):
```

Comment:
It feels weird to have to use the stack context like this. Can you explain why this is necessary?

Reply:
Context variables are essentially thread-safe, stack-level "globals". Think of the Redux / We used to have an entirely different cattrs Converter instance that would strip defaults. With GenConverter, we now have the ability to use the same converter while using an unstruc hook factory to control whether or not to strip defaults based on the value of `ShouldStripDefaults` in the current call stack. The `stack_context` context manager ensures that this value is set only inside the `with` block and gets reverted back to its default when exiting the block, even if an uncaught exception is thrown inside it.